include/serdes.h

```diff
@@ -0,0 +1,71 @@
+#ifndef __UVWASI_SERDES_H__
+#define __UVWASI_SERDES_H__
+
+#include "wasi_types.h"
+
+//
+// Basic uint{8,16,32,64}_t read/write functions.
+//
+
+#define BASIC_TYPE(type)                                                       \
+  void uvwasi_serdes_write_##type(void* ptr, size_t offset, type value);       \
+  type uvwasi_serdes_read_##type(const void* ptr, size_t offset);              \
+
+#define UVWASI_SERDES_SIZE_uint8_t sizeof(uint8_t)
+BASIC_TYPE(uint8_t)
+#define UVWASI_SERDES_SIZE_uint16_t sizeof(uint16_t)
+BASIC_TYPE(uint16_t)
+#define UVWASI_SERDES_SIZE_uint32_t sizeof(uint32_t)
+BASIC_TYPE(uint32_t)
+#define UVWASI_SERDES_SIZE_uint64_t sizeof(uint64_t)
+BASIC_TYPE(uint64_t)
+
+#undef BASIC_TYPE
+
+//
+// WASI structure read/write functions.
```

Comment:
They are guaranteed to be scalar types, though, right? In that case I would add them as `BASIC_TYPE` (so that they return the value instead of taking a pointer to a mutable value).